### PR TITLE
feat: LSDV-4702: Updating vulnerable deps and fixing cross depencies

### PR DESCRIPTION
**Pull request by wesleylima**
### PR fulfills these requirements
- [ ] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [ ] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [ ] Frontend

### Describe the reason for change
_(link to issue, supportive screenshots etc.)_

#### What does this fix?
_(if this is a bug fix)_

#### What is the new behavior?
_(if this is a breaking or feature change)_

#### What is the current behavior?
_(if this is a breaking or feature change)_

#### What libraries were added/updated?
wheel
google-api-core
google-auth
rpc-google-iam-v1
Django
pytz
redis

#### Does this change affect performance?
_(if so describe the impacts positive or negative)_

#### Does this change affect security?
Upgrades packages to address security vulnerabilities

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
_(briefly list any if applicable)_

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit
Regression tests

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

### CODE REVIEW

File: deploy/requirements.txt [hunk 1]
```diff
@@ -50,15 +50,15 @@ protobuf>=3.15.5
...
+pytz ~=2022.1
 requests>=2.22.0,<=2.28
 rq==1.10.1
 rules==2.2
 ujson>=3.0.0
 xmljson==0.2.0
 colorama>=0.4.4
 boxing>=0.1.4
-redis>=4.2.0
+redis~=3.5
```
Comment (bmartel) [inline deploy/requirements.txt:61]: Why the downgrade?

Comment (wesleylima) [reply deploy/requirements.txt:61]: django-redis-cache required in LSE requires Redis < 4 https://github.com/sebleier/django-redis-cache/blob/master/requirements.txt

Comment (bmartel) [reply deploy/requirements.txt:61]: ahh that is what constantly downgrades it when run manually locally.

**Review by bmartel — APPROVED** [review]

File: deploy/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-wheel
+wheel>=0.38.1
```
Comment (makseq) [inline deploy/requirements.txt]: please, don't use too broad dependencies. it lead to deps hell. 
we can limit it like `wheel>=0.38.1,<=MAX_AVAILABLE_AND_WORKING_AT_THIS_MOMENT VERSION`

File: deploy/requirements.txt (hunk 1 above)
Comment (makseq) [inline deploy/requirements.txt:61]: why do we need to touch it? it can affect users who install LS ML backend in the same env as LS (they like to do it).

**Review by makseq — CHANGES_REQUESTED** [review]

File: deploy/requirements.txt (hunk 1 above)
Comment (wesleylima) [reply deploy/requirements.txt:61]: This redis requirement shouldn't conflict with ML backend either. https://github.com/heartexlabs/label-studio-ml-backend/blob/cb98234e3afb6bae2c8ca8a35941a73aceaf3ef8/requirements.txt#L13 
`redis<4.6.0` and `redis>=4.2.0` don't conflict.
Is there another repo's dependencies that we need to be thinking about?

**Review by makseq — APPROVED** [review]

**Review by makseq — APPROVED** [review]

**Pull request merged**